src/mpi/comm/comm_split_type.c

```diff
@@ -106,79 +106,47 @@ int MPIR_Comm_split_type_node(MPIR_Comm * user_comm_ptr, int split_type, int key
 
 #ifdef HAVE_HWLOC
 
-struct info_obj {
-    hwloc_obj_type_t obj_type;
-    union {
-        struct {
-            hwloc_obj_cache_type_t cache_type;
-            int depth;
-        } cache;
-    } u;
-};
-
-struct shmem_processor_info_table {
+struct shmem_info_table {
```

Comment:
This change seems orthogonal to the `hwloc-2.0` upgrade.

Reply:
sorry, will fix this.